### PR TITLE
Add GetOAuthToken method

### PR DESCRIPTION
**Pull request by Hades32**
## Type of change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 🌟 New feature (non-breaking change which adds functionality)
- [ ] 🔨 Breaking change (fix or feature that would cause existing functionality)
- [ ] 📖 Docs change / refactoring / dependency upgrade to change)

## Description

Adds a method for the `oauth_access_tokens` endpoint

### Related Issue (optional)

Relates to #72

### CODE REVIEW

File: clerk/users.go [hunk 1]
```diff
@@ -249,6 +249,27 @@ func (s *UsersService) DisableMFA(userID string) (*UserDisableMFAResponse, error
 	return &response, nil
 }
 
+type OAuthProviderToken struct {
+	Object         string      `json:"object"`
+	Provider       string      `json:"provider"`
+	Token          string      `json:"token"`
+	PublicMetadata interface{} `json:"public_metadata"`
+	Label          string      `json:"label"`
```
Comment (gkats) [inline clerk/users.go:257]: The `label` attribute is optional, let's make it a pointer to a `string` instead.
```suggestion
	Label          *string      `json:"label"`
```

File: clerk/users.go [hunk 2]
```diff
@@ -249,6 +249,27 @@ func (s *UsersService) DisableMFA(userID string) (*UserDisableMFAResponse, error
 	return &response, nil
 }
 
+type OAuthProviderToken struct {
+	Object         string      `json:"object"`
+	Provider       string      `json:"provider"`
+	Token          string      `json:"token"`
+	PublicMetadata interface{} `json:"public_metadata"`
```
Comment (gkats) [inline clerk/users.go:256]: I think the `json.RawMessage` type communicates intent better than the `interface{}` type.
```suggestion
	PublicMetadata json.RawMessage `json:"public_metadata"`
```

File: clerk/users.go [hunk 3]
```diff
@@ -249,6 +249,27 @@ func (s *UsersService) DisableMFA(userID string) (*UserDisableMFAResponse, error
...
+	Label          string      `json:"label"`
+	Scopes         []string    `json:"scopes"`
```
Comment (gkats) [inline clerk/users.go:258]: Turns out that the `scopes` property is available only OAuth 2.0 tokens. Oauth 1.0 tokens use a `token_secret` property.
```suggestion
        // Only set in OAuth 2.0 tokens
	Scopes []string    `json:"scopes,omitempty"`
	// Only set in OAuth 1.0 tokens
	TokenSecret string      `json:"token_secret,omitempty"`
```

File: clerk/users.go [hunk 4]
```diff
@@ -249,6 +249,27 @@ func (s *UsersService) DisableMFA(userID string) (*UserDisableMFAResponse, error
...
+	url := fmt.Sprintf("%s/%s/oauth_access_tokens/%s", UsersUrl, userID, providerID)
+	req, _ := s.client.NewRequest(http.MethodGet, url)
+
+	var response []OAuthProviderToken
+	if _, err := s.client.Do(req, &response); err != nil {
+		return nil, err
+	}
+
+	return response, nil
+}
```
Comment (gkats) [inline clerk/users.go:271]: @Hades32 Could you please add a unit test for this method in `clerk/users_test.go`? Thanks!

File: clerk/clerk.go [hunk 5]
```diff
@@ -213,12 +213,7 @@ func (c *client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	}
 
 	if resp.Body != nil && v != nil {
-		body, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return resp, err
-		}
-
-		err = json.Unmarshal(body, &v)
+		err = json.NewDecoder(resp.Body).Decode(&v)
```
Comment (gkats) [inline clerk/clerk.go:216]: :100:

**Review by gkats — CHANGES_REQUESTED** [review]
Hey @Hades32, huge thanks for the PR! :raised_hands: 

Overall looks great, I've left a few comments regarding the response type. 

Would you be willing to add a unit test for the new method before we can merge this? Thanks!

**Comment by SokratisVidros** [comment]
Closing this, due to the addition of [ListOAuthAccessTokens method](https://github.com/clerkinc/clerk-sdk-go/blob/main/clerk/users.go#L249)

**Pull request closed without merging**